### PR TITLE
Showmap - Pull request

### PR DESCRIPTION
**Pull request by sankarsiva007**
Hi Shanmukh, as informed earlier, I've completed the showmap class but I had to make some changes in MapLoader.java (#68) and MapHelper.java (#14) . 
Hence can you please review the same to make sure it's not impacting other functionalities and complete the pull request.
Please let me know if there is anything required.
Thanks!

### CODE REVIEW

File: src/main/java/game/map/MapLoader.java [hunk 1]
```diff
@@ -63,13 +65,26 @@ public static void loadMap(String path, Map map) {
...
 
+//                if (readingBorders) {
+//                    String[] borderAttributes = line.split(" ");
+//                    Country currentCountry = getCountryWithId(map, Integer.parseInt(borderAttributes[0]));
+//                    for (int idx = 1; idx < borderAttributes.length; idx ++) {
+//                        int neighbourId = Integer.parseInt(borderAttributes[idx]);
+//                        Country neighbour = getCountryWithId(map, neighbourId);
+//                        currentCountry.getNeighbours().add(neighbour);
+//                    }
+//                }
```
Comment (shanmukhatummala) [inline src/main/java/game/map/MapLoader.java:76]: commented code could be removed if not needed.

File: src/main/java/game/map/ShowMap.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        System.out.println("---------------------------Continents---------------------------");
+        List<Continent> continents = map.getContinents();
+        for (Continent continent : continents) {
+            System.out.println(continent.getName());
+        }
+        System.out.println("---------------------------Countries---------------------------");
+        List<Country> countries = map.getCountries();
+        for (Country country : countries) {
+            System.out.println(country.getName());
+            System.out.println("Borders:");
```
Comment (shanmukhatummala) [inline src/main/java/game/map/ShowMap.java:16]: Why to use the term borders?
(as they used neighbors in the description - https://users.encs.concordia.ca/~paquet/wiki/images/6/6f/Build1Grading.SOEN6441.2024.4.pdf)

File: src/main/java/game/pojo/Country.java [hunk 3]
```diff
@@ -11,19 +11,21 @@ public class Country {
     List<Country> neighbours;
     Player player;
     int armyCount;
+    List<Integer> borderIds;
```
Comment (shanmukhatummala) [inline src/main/java/game/pojo/Country.java:14]: This need not be present as we already have `List<Country> neighbours;`. So you can remove all the changes made in Country.java and just use neighbors in your code in Showmap.java.

File: src/main/java/game/map/ShowMap.java (hunk 2 above)
Comment (shanmukhatummala) [reply src/main/java/game/map/ShowMap.java:16]: You can use the below to print all the neighboring countries to a country.

```suggestion
            for (Country country : countries) {
                System.out.println(country.getName());
                System.out.println("Neighbors:");
                for (Country neighbor : country.getNeighbors) {
                    System.out.println("- " + neighbor.getName());
                }
            }
```

File: src/main/java/game/map/MapHelper.java [hunk 4]
```diff
@@ -11,9 +11,20 @@ public static Continent getContinentWithId(Map map, int id) {
...
+// Below method throws IndexOutOfBoundsException when no country is found with the given id. But the next method
+// will return "null" which seems to be a safer option. Leaving it to Shanmukh for confirmation as the initial method
+// was created by him.
 
+//    public static Country getCountryWithId(Map map, int id) {
+//
+//        return map.getCountries()
+//                .stream().filter(country -> country.getId() == id).toList().get(0);
+//    }
+    public static Country getCountryWithId(Map map, int id) {
```
Comment (shanmukhatummala) [inline src/main/java/game/map/MapHelper.java:23]: Yes, this is a better way of doing things. But people who call this method have to be aware that if they use an Id that doesn't exist, then they would get null as the response and handle it properly whenever they call the method. (whereas previously if you use a bad Id, an exception will occur and things will fail immediately).

Could you remove the code you commented and update `getContinentWithId` the same way?

Also, please check whoever used these methods handled things properly when this method return null. You can also add the code to handle the case when this method returns `null`, wherever this method is called.

File: src/main/java/game/map/MapLoader.java
Comment (shanmukhatummala) [inline src/main/java/game/map/MapLoader.java:1]: Please don't make any changes to this file. Check the comments in Country.java and ShowMap.java.

File: src/main/java/game/map/ShowMap.java
Comment (shanmukhatummala) [inline src/main/java/game/map/ShowMap.java:1]: Minor suggestion - Naming this class as MapShower would be great, to keep it similar to other classes names.

File: src/main/java/game/GameEngine.java [hunk 5]
```diff
@@ -40,8 +41,13 @@ public static void main(String[] args) {
                         createNewMap(fileName);
                     }
                     loadMap(path, map);
-                    endGame();
-                } else {
+                    // endGame();
+                }
+                else if (commandArgs.length == 1 && "showmap".equals(commandArgs[0])) {
+                    Map Map;
```
Comment (shanmukhatummala) [inline src/main/java/game/GameEngine.java:47]: Don't see any use of `Map Map`. This line should be removed.

**Review by shanmukhatummala — COMMENTED** [review]
Added comments.

File: src/main/java/game/GameEngine.java (hunk 5 above)
Comment (sankarsiva007) [reply src/main/java/game/GameEngine.java:47]: Done

File: src/main/java/game/map/ShowMap.java
Comment (sankarsiva007) [reply src/main/java/game/map/ShowMap.java:1]: Done

File: src/main/java/game/pojo/Country.java (hunk 3 above)
Comment (sankarsiva007) [reply src/main/java/game/pojo/Country.java:14]: Done

**Pull request closed without merging**